src/v/bytes/details/io_byte_iterator.h

```diff
@@ -72,6 +72,11 @@ class io_byte_iterator {
         }
         return *this;
     }
+    io_byte_iterator operator++(int) {
```

Comment:
Curious what the int is for? Is it wrong to assume that the caller expects this to increment N times? Or is that just not safe because we expect the caller to be checking against end

Reply:
i wonder if they agonized over the decision of what type the dummy parameter should be